crates/amaru-consensus/src/consensus/tx_submission/tests/tx_submission_client.rs

```diff
@@ -0,0 +1,311 @@
+// Copyright 2025 PRAGMA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::consensus::tx_submission::TxSubmissionClientState;
+use crate::consensus::tx_submission::tests::{
+    MockClientTransport, SizedMempool, TxClientTransport, assert_next_message,
+    assert_tx_bodies_reply, assert_tx_ids_reply, create_transactions,
+};
+use crate::consensus::tx_submission::tx_submission_client_state::TxClientResponse;
+use amaru_kernel::peer::Peer;
+use amaru_network::{era_tx_bodies, era_tx_id, era_tx_ids, to_pallas_request};
+use amaru_ouroboros_traits::{Mempool, TxId, TxSubmissionMempool};
+use pallas_network::miniprotocols::txsubmission::{EraTxBody, EraTxId, Message, TxIdAndSize};
+use pallas_primitives::conway::Tx;
+use std::sync::Arc;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::{Receiver, Sender};
+use tokio::task::JoinHandle;
+use tracing::{debug, info};
+
+/// Tx submission client state machine for a given peer.
+/// Most of the logic is in the `TxSubmissionClientState`,
+/// this struct just wires it up to a mempool and a transport.
+#[derive(Clone)]
+pub struct TxSubmissionClient<Tx> {
+    /// Mempool to pull transactions from.
+    mempool: Arc<dyn TxSubmissionMempool<Tx>>,
+    /// In-memory state of the client.
+    state: TxSubmissionClientState,
+}
+
+impl TxSubmissionClient<Tx> {
+    /// Create a new tx submission client state machine for the given mempool and peer.
+    pub fn new(mempool: Arc<dyn TxSubmissionMempool<Tx>>, peer: &Peer) -> Self {
+        Self {
+            mempool: mempool.clone(),
+            state: TxSubmissionClientState::new(peer),
+        }
+    }
+
+    /// Core state machine, parameterized over a transport for testability.
+    pub async fn start_client_with_transport<T: TxClientTransport>(
+        &mut self,
```

Comment:
My removing this `&` and requiring `T: Send + 'static` you could directly spawn the resulting future without wrapping in `async move { ….await }`